lib/parser.js

```diff
@@ -159,13 +159,21 @@ async function customDocumentOperations(js, asyncapiYAMLorJSON, initialFormat, o
     const convert = OPERATIONS.map(async (opName) => {
       const op = channel[opName];
       if (op) {
+        let messages = [];
+        const msg = op.message;
+        if (msg) {
+          if (msg.oneOf) {
+            messages = msg.oneOf;
+          } else {
+            messages = [msg];
+          }
+        }
+
         if (options.applyTraits) {  
           applyTraits(op);
-          applyTraits(op.message);
+          messages.forEach(m => applyTraits(m));
         }
-        
-        const msg = op.message;
-        if (msg) await validateAndConvertMessage(msg);
+        messages.forEach(m => validateAndConvertMessage(m)); 
```

Comment:
Since `validateAndConvertMessage` is async `forEach` is not possible to use here is it? 🤔

Reply:
I missed it. I will correct it. Thanks for review! :)